fhirstore/fhirstore.py

```diff
@@ -290,75 +289,84 @@ def search(
         """
         self.validate_resource_type(resource_type)
 
-        core_query = build_core_query(params)
-        query = {
-            "min_score": 0.01,
-            "from": offset,
-            "size": result_size,
-            "query": core_query,
-        }
-
-        if sort:
-            query["sort"] = sort
-
-
-        if elements:
-            query["_source"] = elements
-
-        # .lower() is used to fix the fact that monstache changes resourceTypes to
-        # all lower case
-        hits = self.es.search(body=query, index=f"fhirstore.{resource_type.lower()}")
-        bundle = {
-            "resource_type": "Bundle",
-            "items": [
-                {"resource": h["_source"], "search": {"mode": "match"}}
-                for h in hits["hits"]["hits"]
-            ],
-
-            "total": hits["hits"]["total"]["value"],
-        }
-
-        if elements:
-            bundle["tag"] = {"code": "SUBSETTED"}
-
-        if include:
-            #For each result instance
-            for item in bundle["items"]:
-                #For each attribute to include
-                for attribute in include:
-                    # split the reference attribute "Practioner/123" into a 
-                    # resource "Practioner" and an id "123"
-                    included_resource, included_id = re.split(
-                        "\/", item[attribute]["reference"], maxsplit=1
-                    )
-                    # search the db for the specific resource to include
-                    included_hits = self.es.search(
-                        body={"simple_query_string": {"query": included_id, "fields": "id",}},
-                        index=f"fhirstore.{included_resource.lower()}",
-                    )
-            [
-                bundle["items"].append({"resource": h["_source"], "search": {"mode": "include"}})
-                for h in included_hits["hits"]["hits"]
-            ]
-
-        return bundle
-
-    def count(self, resource_type, params):
-        """Counts how many results match this query
-        """
-        self.validate_resource_type(resource_type)
+        # create a Parser instance and process the arguments
+        parsed_args = URL_Parser(args, resource_type)
```

Comment:
I think it would be better to call the URLParser instance `parser` or something like that.
And maybe the method `process_params()` could return the parse args. We could then have something like that here:
```
        parser = URL_Parser(args, resource_type)

        parsed_args = parser.process_params()

        # Create an instance of querybuilder and build core query
        core_query = CoreQueryBuilder(parsed_args)
```

Reply:
I put process_params() inside the __init__ function of the URL_Parser class, that way there's no need to call it twice.